model/github/github_repository.rb

```diff
@@ -1,14 +1,38 @@
 # frozen_string_literal: true
 
+require "aws-sdk-s3"
+
 require_relative "../../model"
 
 class GithubRepository < Sequel::Model
   one_to_one :strand, key: :id
   many_to_one :installation, key: :installation_id, class: :GithubInstallation
   one_to_many :runners, key: :repository_id, class: :GithubRunner
+  one_to_many :cache_entries, key: :repository_id, class: :GithubCacheEntry
 
   include ResourceMethods
   include SemaphoreMethods
 
   semaphore :destroy
+
+  plugin :column_encryption do |enc|
+    enc.column :secret_key
+  end
+
+  def bucket_name
+    ubid
+  end
+
+  def blob_storage_client
+    @blob_storage_client ||= Aws::S3::Client.new(
+      endpoint: Config.github_cache_blob_storage_endpoint,
+      access_key_id: access_key,
+      secret_access_key: secret_key,
```

Comment:
> This PR lack some tests and one small part where we create a bucket
Sorry for the confusion. This relates to the part where we create a bucket.

The values from ﻿`Config` are the admin credentials, which enable us to create a bucket and generate an access key for it.

The values from the repository columns are the credentials needed to access a specific bucket. We set these when we actually create the bucket.

Reply:
I updated the commit messages. Hopefully, they're clearer now.

Reply:
Got it, but do you think we need separate secrets for each bucket, given that they are all generated, stored and used exclusively in the control plane. Only the presigned url is being passed to the data plane, which can be generated from single source.

Reply:
The pre-signed URL contains raw `access_key` (so this information is not private), and a `secret_key` is used to generate the signature. I'm not sure how it happens, but if the `secret_key` for a repository leak somehow, it would only affect that single repository, not all repositories.